TST: theilsope & siegelslope-related tests are failing on PyPy3.11 7.3.19: `TypeError: tuple() expected at most 1 argument, got 2`

### Describe your issue.

It seems that something is going wrong with `result_to_tuple` handling but I couldn't figure out what's happening exactly. See the example below.

### Reproducing Code Example

```python
pypy3.11 -m pytest scipy/stats/tests/test_mstats_basic.py::test_siegelslopes
```

### Error message

```shell
============================= test session starts ==============================
platform linux -- Python 3.11.11[pypy-7.3.19-final], pytest-8.3.5, pluggy-1.6.0
rootdir: /var/tmp/portage/dev-python/scipy-1.16.0_rc1/work/scipy-1.16.0rc1-pypy3_11/install/usr/lib/pypy3.11/site-packages
collected 1 item

scipy/stats/tests/test_mstats_basic.py F                                 [100%]

=================================== FAILURES ===================================
______________________________ test_siegelslopes _______________________________

    def test_siegelslopes():
        # method should be exact for straight line
        y = 2 * np.arange(10) + 0.5
>       assert_equal(mstats.siegelslopes(y), (2.0, 0.5))

scipy/stats/tests/test_mstats_basic.py:1025: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
scipy/stats/_mstats_basic.py:1313: in siegelslopes
    return stats_siegelslopes(y, x, method=method)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

_no_deco = False, kwds = {'method': 'hierarchical'}
temp = array([ 0.5,  2.5,  4.5,  6.5,  8.5, 10.5, 12.5, 14.5, 16.5, 18.5])
params = ['y', 'x', 'method'], maxarg = 3
d_args = {'x': array([0., 1., 2., 3., 4., 5., 6., 7., 8., 9.]), 'y': array([ 0.5,  2.5,  4.5,  6.5,  8.5, 10.5, 12.5, 14.5, 16.5, 18.5])}
intersection = set()
samples = [array([ 0.5,  2.5,  4.5,  6.5,  8.5, 10.5, 12.5, 14.5, 16.5, 18.5]), array([0., 1., 2., 3., 4., 5., 6., 7., 8., 9.])]
vectorized = False

    @wraps(hypotest_fun_in)
    def axis_nan_policy_wrapper(*args, _no_deco=False, **kwds):
    
        if _no_deco:  # for testing, decorator does nothing
            return hypotest_fun_in(*args, **kwds)
    
        # For now, skip the decorator entirely if using array API. In the future,
        # we'll probably want to use it for `keepdims`, `axis` tuples, etc.
        if len(args) == 0:  # extract sample from `kwds` if there are no `args`
            used_kwd_samples = list(set(kwds).intersection(set(kwd_samples)))
            temp = used_kwd_samples[:1]
        else:
            temp = args[0]
    
        if not is_numpy(array_namespace(temp)):
            msg = ("Use of `nan_policy` and `keepdims` "
                   "is incompatible with non-NumPy arrays.")
            if 'nan_policy' in kwds or 'keepdims' in kwds:
                raise NotImplementedError(msg)
            return hypotest_fun_in(*args, **kwds)
    
        # We need to be flexible about whether position or keyword
        # arguments are used, but we need to make sure users don't pass
        # both for the same parameter. To complicate matters, some
        # functions accept samples with *args, and some functions already
        # accept `axis` and `nan_policy` as positional arguments.
        # The strategy is to make sure that there is no duplication
        # between `args` and `kwds`, combine the two into `kwds`, then
        # the samples, `nan_policy`, and `axis` from `kwds`, as they are
        # dealt with separately.
    
        # Check for intersection between positional and keyword args
        params = list(inspect.signature(hypotest_fun_in).parameters)
        if n_samples is None:
            # Give unique names to each positional sample argument
            # Note that *args can't be provided as a keyword argument
            params = [f"arg{i}" for i in range(len(args))] + params[1:]
    
        # raise if there are too many positional args
        maxarg = (np.inf if inspect.getfullargspec(hypotest_fun_in).varargs
                  else len(inspect.getfullargspec(hypotest_fun_in).args))
        if len(args) > maxarg:  # let the function raise the right error
            hypotest_fun_in(*args, **kwds)
    
        # raise if multiple values passed for same parameter
        d_args = dict(zip(params, args))
        intersection = set(d_args) & set(kwds)
        if intersection:  # let the function raise the right error
            hypotest_fun_in(*args, **kwds)
    
        # Consolidate other positional and keyword args into `kwds`
        kwds.update(d_args)
    
        # rename avoids UnboundLocalError
        if callable(n_samples):
            # Future refactoring idea: no need for callable n_samples.
            # Just replace `n_samples` and `kwd_samples` with a single
            # list of the names of all samples, and treat all of them
            # as `kwd_samples` are treated below.
            n_samp = n_samples(kwds)
        else:
            n_samp = n_samples or len(args)
    
        # get the number of outputs
        n_out = n_outputs  # rename to avoid UnboundLocalError
        if callable(n_out):
            n_out = n_out(kwds)
    
        # If necessary, rearrange function signature: accept other samples
        # as positional args right after the first n_samp args
        kwd_samp = [name for name in kwd_samples
                    if kwds.get(name, None) is not None]
        n_kwd_samp = len(kwd_samp)
        if not kwd_samp:
            hypotest_fun_out = hypotest_fun_in
        else:
            def hypotest_fun_out(*samples, **kwds):
                new_kwds = dict(zip(kwd_samp, samples[n_samp:]))
                kwds.update(new_kwds)
                return hypotest_fun_in(*samples[:n_samp], **kwds)
    
        # Extract the things we need here
        try:  # if something is missing
            samples = [np.atleast_1d(kwds.pop(param))
                       for param in (params[:n_samp] + kwd_samp)]
        except KeyError:  # let the function raise the right error
            # might need to revisit this if required arg is not a "sample"
            hypotest_fun_in(*args, **kwds)
        vectorized = True if 'axis' in params else False
        vectorized = vectorized and not override['vectorization']
        axis = kwds.pop('axis', default_axis)
        nan_policy = kwds.pop('nan_policy', 'propagate')
        keepdims = kwds.pop("keepdims", False)
        del args  # avoid the possibility of passing both `args` and `kwds`
    
        # convert masked arrays to regular arrays with sentinel values
        samples, sentinel = _masked_arrays_2_sentinel_arrays(samples)
    
        # standardize to always work along last axis
        reduced_axes = axis
        if axis is None:
            if samples:
                # when axis=None, take the maximum of all dimensions since
                # all the dimensions are reduced.
                n_dims = np.max([sample.ndim for sample in samples])
                reduced_axes = tuple(range(n_dims))
            samples = [np.asarray(sample.ravel()) for sample in samples]
        else:
            # don't ignore any axes when broadcasting if paired
            samples = _broadcast_arrays(samples, axis=axis if not paired else None)
            axis = np.atleast_1d(axis)
            n_axes = len(axis)
            # move all axes in `axis` to the end to be raveled
            samples = [np.moveaxis(sample, axis, range(-len(axis), 0))
                       for sample in samples]
            shapes = [sample.shape for sample in samples]
            # New shape is unchanged for all axes _not_ in `axis`
            # At the end, we append the product of the shapes of the axes
            # in `axis`. Appending -1 doesn't work for zero-size arrays!
            new_shapes = [shape[:-n_axes] + (np.prod(shape[-n_axes:]),)
                          for shape in shapes]
            samples = [sample.reshape(new_shape)
                       for sample, new_shape in zip(samples, new_shapes)]
        axis = -1  # work over the last axis
        NaN = _get_nan(*samples) if samples else np.nan
    
        # if axis is not needed, just handle nan_policy and return
        ndims = np.array([sample.ndim for sample in samples])
        if np.all(ndims <= 1):
            # Addresses nan_policy == "raise"
            if nan_policy != 'propagate' or override['nan_propagation']:
                contains_nan = [_contains_nan(sample, nan_policy)
                                for sample in samples]
            else:
                # Behave as though there are no NaNs (even if there are)
                contains_nan = [False] * len(samples)
    
            # Addresses nan_policy == "propagate"
            if any(contains_nan) and (nan_policy == 'propagate'
                                      and override['nan_propagation']):
                res = np.full(n_out, NaN)
                res = _add_reduced_axes(res, reduced_axes, keepdims)
                return tuple_to_result(*res)
    
            # Addresses nan_policy == "omit"
            too_small_msg = too_small_1d_not_omit
            if any(contains_nan) and nan_policy == 'omit':
                # consider passing in contains_nan
                samples = _remove_nans(samples, paired)
                too_small_msg = too_small_1d_omit
    
            if sentinel:
                samples = _remove_sentinel(samples, paired, sentinel)
    
            if is_too_small(samples, kwds):
                warnings.warn(too_small_msg, SmallSampleWarning, stacklevel=2)
                res = np.full(n_out, NaN)
                res = _add_reduced_axes(res, reduced_axes, keepdims)
                return tuple_to_result(*res)
    
            res = hypotest_fun_out(*samples, **kwds)
>           res = result_to_tuple(res, n_out)
E           TypeError: tuple() expected at most 1 argument, got 2

scipy/stats/_axis_nan_policy.py:588: TypeError
=============================== warnings summary ===============================
scipy/stats/tests/test_mstats_basic.py:395
  /var/tmp/portage/dev-python/scipy-1.16.0_rc1/work/scipy-1.16.0rc1-pypy3_11/install/usr/lib/pypy3.11/site-packages/scipy/stats/tests/test_mstats_basic.py:395: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.slow

scipy/stats/tests/test_mstats_basic.py:463
  /var/tmp/portage/dev-python/scipy-1.16.0_rc1/work/scipy-1.16.0rc1-pypy3_11/install/usr/lib/pypy3.11/site-packages/scipy/stats/tests/test_mstats_basic.py:463: PytestUnknownMarkWarning: Unknown pytest.mark.xslow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.xslow

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED scipy/stats/tests/test_mstats_basic.py::test_siegelslopes - TypeError:...
======================== 1 failed, 2 warnings in 1.13s =========================
```

### SciPy/NumPy/Python version and system information

```shell
1.16.0rc1 2.2.6 sys.version_info(major=3, minor=11, micro=11, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include
    lib directory: /usr/lib64
    name: blas
    openblas configuration: unknown
    pc file directory: /usr/lib64/pkgconfig
    version: 3.12.0
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include
    lib directory: /usr/lib64
    name: lapack
    openblas configuration: unknown
    pc file directory: /usr/lib64/pkgconfig
    version: 3.12.0
  pybind11:
    detection method: pkgconfig
    include directory: /include
    name: pybind11
    version: 3.0.0
Compilers:
  c:
    args: -march=znver2, --param=l1-cache-size=32, --param=l1-cache-line-size=64,
      -O2, -pipe, -frecord-gcc-switches, -UNDEBUG
    commands: x86_64-pc-linux-gnu-gcc
    linker: ld.bfd
    linker args: -march=znver2, --param=l1-cache-size=32, --param=l1-cache-line-size=64,
      -O2, -pipe, -frecord-gcc-switches, -Wl,-O1, -Wl,--as-needed, -Wl,--defsym=__gentoo_check_ldflags__=0
    name: gcc
    version: 14.2.1
  c++:
    args: -march=znver2, --param=l1-cache-size=32, --param=l1-cache-line-size=64,
      -O2, -pipe, -frecord-gcc-switches, -UNDEBUG
    commands: x86_64-pc-linux-gnu-g++
    linker: ld.bfd
    linker args: -march=znver2, --param=l1-cache-size=32, --param=l1-cache-line-size=64,
      -O2, -pipe, -frecord-gcc-switches, -Wl,-O1, -Wl,--as-needed, -Wl,--defsym=__gentoo_check_ldflags__=0
    name: gcc
    version: 14.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.1.1
  fortran:
    args: -march=znver2, --param=l1-cache-size=32, --param=l1-cache-line-size=64,
      -O2, -pipe, -frecord-gcc-switches
    commands: x86_64-pc-linux-gnu-gfortran
    linker: ld.bfd
    linker args: -march=znver2, --param=l1-cache-size=32, --param=l1-cache-line-size=64,
      -O2, -pipe, -frecord-gcc-switches, -Wl,-O1, -Wl,--as-needed, -Wl,--defsym=__gentoo_check_ldflags__=0
    name: gcc
    version: 14.2.1
  pythran:
    include directory: ../../../../../../../../usr/lib/pypy3.11/site-packages/pythran
    version: 0.18.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /usr/bin/pypy3.11
  version: '3.11'
```

--- a/scipy/stats/_axis_nan_policy.py
+++ b/scipy/stats/_axis_nan_policy.py
@@ -403,16 +403,8 @@ def _axis_nan_policy_factory(tuple_to_result, default_axis=0,
     override.update(temp)
 
     if result_to_tuple is None:
-        def result_to_tuple(res):
+        def result_to_tuple(res, _):
             return res
-
-    # The only `result_to_tuple` that needs the second argument (number of
-    # outputs) is the one for `moment`, and this was realized very late.
-    # Rather than changing all `result_to_tuple` definitions, we wrap them
-    # here to accept a second argument if they don't already.
-    if len(inspect.signature(result_to_tuple).parameters) == 1:
-        def result_to_tuple(res, _, f=result_to_tuple):
-            return f(res)
 
     if not callable(too_small):
         def is_too_small(samples, *ts_args, axis=-1, **ts_kwargs):

--- a/scipy/stats/_correlation.py
+++ b/scipy/stats/_correlation.py
@@ -81,7 +81,7 @@ def _chatterjeexi_iv(y_continuous, method):
     return y_continuous, method
 
 
-def _unpack(res):
+def _unpack(res, _):
     return res.statistic, res.pvalue
 
 

--- a/scipy/stats/_entropy.py
+++ b/scipy/stats/_entropy.py
@@ -21,7 +21,7 @@ __all__ = ['entropy', 'differential_entropy']
         2 if ("qk" in kwgs and kwgs["qk"] is not None)
         else 1
     ),
-    n_outputs=1, result_to_tuple=lambda x: (x,), paired=True,
+    n_outputs=1, result_to_tuple=lambda x, _: (x,), paired=True,
     too_small=-1  # entropy doesn't have too small inputs
 )
 def entropy(pk: np.typing.ArrayLike,
@@ -176,7 +176,7 @@ def _differential_entropy_is_too_small(samples, kwargs, axis=-1):
 
 
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,),
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x, _: (x,),
     too_small=_differential_entropy_is_too_small
 )
 def differential_entropy(

--- a/scipy/stats/_hypotests.py
+++ b/scipy/stats/_hypotests.py
@@ -482,7 +482,7 @@ def _cdf_cvm(x, n=None):
     return y
 
 
-def _cvm_result_to_tuple(res):
+def _cvm_result_to_tuple(res, _):
     return res.statistic, res.pvalue
 
 

--- a/scipy/stats/_morestats.py
+++ b/scipy/stats/_morestats.py
@@ -226,7 +226,7 @@ def mvsdist(data):
 
 
 @_axis_nan_policy_factory(
-    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1, default_axis=None
+    lambda x: x, result_to_tuple=lambda x, _: (x,), n_outputs=1, default_axis=None
 )
 def kstat(data, n=2, *, axis=None):
     r"""
@@ -331,7 +331,7 @@ def kstat(data, n=2, *, axis=None):
 
 
 @_axis_nan_policy_factory(
-    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1, default_axis=None
+    lambda x: x, result_to_tuple=lambda x, _: (x,), n_outputs=1, default_axis=None
 )
 def kstatvar(data, n=2, *, axis=None):
     r"""Return an unbiased estimator of the variance of the k-statistic.
@@ -988,7 +988,7 @@ def boxcox_llf(lmb, data, *, axis=0, keepdims=False, nan_policy='propagate'):
 
 
 @_axis_nan_policy_factory(lambda x: x, n_outputs=1, default_axis=0,
-                          result_to_tuple=lambda x: (x,))
+                          result_to_tuple=lambda x, _: (x,))
 def _boxcox_llf(data, axis=0, *, lmb):
     xp = array_namespace(data)
     lmb, data = xp_promote(lmb, data, force_floating=True, xp=xp)
@@ -3505,7 +3505,7 @@ def mood(x, y, axis=0, alternative="two-sided"):
 WilcoxonResult = _make_tuple_bunch('WilcoxonResult', ['statistic', 'pvalue'])
 
 
-def wilcoxon_result_unpacker(res):
+def wilcoxon_result_unpacker(res, _):
     if hasattr(res, 'zstatistic'):
         return res.statistic, res.pvalue, res.zstatistic
     else:
@@ -4002,7 +4002,7 @@ def _circfuncs_common(samples, period, xp=None):
 
 @_axis_nan_policy_factory(
     lambda x: x, n_outputs=1, default_axis=None,
-    result_to_tuple=lambda x: (x,)
+    result_to_tuple=lambda x, _: (x,)
 )
 def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
     r"""Compute the circular mean of a sample of angle observations.
@@ -4095,7 +4095,7 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
 
 @_axis_nan_policy_factory(
     lambda x: x, n_outputs=1, default_axis=None,
-    result_to_tuple=lambda x: (x,)
+    result_to_tuple=lambda x, _: (x,)
 )
 def circvar(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
     r"""Compute the circular variance of a sample of angle observations.
@@ -4189,7 +4189,7 @@ def circvar(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
 
 @_axis_nan_policy_factory(
     lambda x: x, n_outputs=1, default_axis=None,
-    result_to_tuple=lambda x: (x,)
+    result_to_tuple=lambda x, _: (x,)
 )
 def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate', *,
             normalize=False):

--- a/scipy/stats/_stats_mstats_common.py
+++ b/scipy/stats/_stats_mstats_common.py
@@ -21,7 +21,8 @@ def _n_samples_optional_x(kwargs):
 
 @_axis_nan_policy_factory(TheilslopesResult, default_axis=None, n_outputs=4,
                           n_samples=_n_samples_optional_x,
-                          result_to_tuple=tuple, paired=True, too_small=1)
+                          result_to_tuple=lambda x, _: tuple(x), paired=True,
+                          too_small=1)
 def theilslopes(y, x=None, alpha=0.95, method='separate'):
     r"""
     Computes the Theil-Sen estimator for a set of points (x, y).
@@ -204,7 +205,8 @@ def _find_repeats(arr):
 
 @_axis_nan_policy_factory(SiegelslopesResult, default_axis=None, n_outputs=2,
                           n_samples=_n_samples_optional_x,
-                          result_to_tuple=tuple, paired=True, too_small=1)
+                          result_to_tuple=lambda x, _: tuple(x), paired=True,
+                          too_small=1)
 def siegelslopes(y, x=None, method="hierarchical"):
     r"""
     Computes the Siegel estimator for a set of points (x, y).

--- a/scipy/stats/_stats_py.py
+++ b/scipy/stats/_stats_py.py
@@ -160,7 +160,7 @@ def _pack_CorrelationResult(statistic, pvalue, correlation):
     return res
 
 
-def _unpack_CorrelationResult(res):
+def _unpack_CorrelationResult(res, _):
     return res.statistic, res.pvalue, res.correlation
 
 
@@ -168,7 +168,7 @@ def _unpack_CorrelationResult(res):
 @xp_capabilities()
 @_axis_nan_policy_factory(
         lambda x: x, n_samples=1, n_outputs=1, too_small=0, paired=True,
-        result_to_tuple=lambda x: (x,), kwd_samples=['weights'])
+        result_to_tuple=lambda x, _: (x,), kwd_samples=['weights'])
 def gmean(a, axis=0, dtype=None, weights=None):
     r"""Compute the weighted geometric mean along the specified axis.
 
@@ -252,7 +252,7 @@ def gmean(a, axis=0, dtype=None, weights=None):
 @xp_capabilities(jax_jit=False, allow_dask_compute=1)
 @_axis_nan_policy_factory(
         lambda x: x, n_samples=1, n_outputs=1, too_small=0, paired=True,
-        result_to_tuple=lambda x: (x,), kwd_samples=['weights'])
+        result_to_tuple=lambda x, _: (x,), kwd_samples=['weights'])
 def hmean(a, axis=0, dtype=None, *, weights=None):
     r"""Calculate the weighted harmonic mean along the specified axis.
 
@@ -353,7 +353,7 @@ def hmean(a, axis=0, dtype=None, *, weights=None):
 @xp_capabilities(jax_jit=False, allow_dask_compute=1)
 @_axis_nan_policy_factory(
         lambda x: x, n_samples=1, n_outputs=1, too_small=0, paired=True,
-        result_to_tuple=lambda x: (x,), kwd_samples=['weights'])
+        result_to_tuple=lambda x, _: (x,), kwd_samples=['weights'])
 def pmean(a, p, *, axis=0, dtype=None, weights=None):
     r"""Calculate the weighted power mean along the specified axis.
 
@@ -634,7 +634,7 @@ def _put_val_to_limits(a, limits, inclusive, val=np.nan, xp=None):
 @xp_capabilities()
 @_axis_nan_policy_factory(
     lambda x: x, n_outputs=1, default_axis=None,
-    result_to_tuple=lambda x: (x,)
+    result_to_tuple=lambda x, _: (x,)
 )
 def tmean(a, limits=None, inclusive=(True, True), axis=None):
     """Compute the trimmed mean.
@@ -689,7 +689,7 @@ def tmean(a, limits=None, inclusive=(True, True), axis=None):
 
 @xp_capabilities()
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,)
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x, _: (x,)
 )
 def tvar(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
     """Compute the trimmed variance.
@@ -749,7 +749,7 @@ def tvar(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
 
 @xp_capabilities()
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,)
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x, _: (x,)
 )
 def tmin(a, lowerlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
     """Compute the trimmed minimum.
@@ -813,7 +813,7 @@ def tmin(a, lowerlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
 
 @xp_capabilities()
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,)
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x, _: (x,)
 )
 def tmax(a, upperlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
     """Compute the trimmed maximum.
@@ -876,7 +876,7 @@ def tmax(a, upperlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
 
 @xp_capabilities()
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,)
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x, _: (x,)
 )
 def tstd(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
     """Compute the trimmed sample standard deviation.
@@ -929,7 +929,7 @@ def tstd(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
 
 @xp_capabilities()
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,)
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x, _: (x,)
 )
 def tsem(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
     """Compute the trimmed standard error of the mean.
@@ -1246,7 +1246,7 @@ def _var(x, axis=0, ddof=0, mean=None, xp=None):
 
 @xp_capabilities(jax_jit=False, allow_dask_compute=2)
 @_axis_nan_policy_factory(
-    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1
+    lambda x: x, result_to_tuple=lambda x, _: (x,), n_outputs=1
 )
 # nan_policy handled by `_axis_nan_policy`, but needs to be left
 # in signature to preserve use as a positional argument
@@ -1347,7 +1347,7 @@ def skew(a, axis=0, bias=True, nan_policy='propagate'):
 
 @xp_capabilities(jax_jit=False, allow_dask_compute=2)
 @_axis_nan_policy_factory(
-    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1
+    lambda x: x, result_to_tuple=lambda x, _: (x,), n_outputs=1
 )
 # nan_policy handled by `_axis_nan_policy`, but needs to be left
 # in signature to preserve use as a positional argument
@@ -2583,7 +2583,7 @@ def obrientransform(*samples):
 
 @xp_capabilities(jax_jit=False, allow_dask_compute=True)
 @_axis_nan_policy_factory(
-    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1, too_small=1
+    lambda x: x, result_to_tuple=lambda x, _: (x,), n_outputs=1, too_small=1
 )
 def sem(a, axis=0, ddof=1, nan_policy='propagate'):
     """Compute standard error of the mean.
@@ -3051,7 +3051,7 @@ _scale_conversions = {'normal': special.erfinv(0.5) * 2.0 * math.sqrt(2.0)}
 
 
 @_axis_nan_policy_factory(
-    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1,
+    lambda x: x, result_to_tuple=lambda x, _: (x,), n_outputs=1,
     default_axis=None, override={'nan_propagation': False}
 )
 def iqr(x, axis=None, rng=(25, 75), scale=1.0, nan_policy='propagate',
@@ -4049,7 +4049,7 @@ class AlexanderGovernResult:
 
 @_axis_nan_policy_factory(
     AlexanderGovernResult, n_samples=None,
-    result_to_tuple=lambda x: (x.statistic, x.pvalue),
+    result_to_tuple=lambda x, _: (x.statistic, x.pvalue),
     too_small=1
 )
 def alexandergovern(*samples, nan_policy='propagate', axis=0):
@@ -6011,7 +6011,7 @@ def pack_TtestResult(statistic, pvalue, df, alternative, standard_error,
                        standard_error=standard_error, estimate=estimate)
 
 
-def unpack_TtestResult(res):
+def unpack_TtestResult(res, _):
     return (res.statistic, res.pvalue, res.df, res._alternative,
             res._standard_error, res._estimate)
 
@@ -7617,7 +7617,7 @@ def _tuple_to_KstestResult(statistic, pvalue,
                         statistic_sign=statistic_sign)
 
 
-def _KstestResult_to_tuple(res):
+def _KstestResult_to_tuple(res, _):
     return *res, res.statistic_location, res.statistic_sign
 
 
@@ -10590,7 +10590,7 @@ def _pack_LinregressResult(slope, intercept, rvalue, pvalue, stderr, intercept_s
                             intercept_stderr=intercept_stderr)
 
 
-def _unpack_LinregressResult(res):
+def _unpack_LinregressResult(res, _):
     return tuple(res) + (res.intercept_stderr,)
 
 

--- a/scipy/stats/_variation.py
+++ b/scipy/stats/_variation.py
@@ -8,7 +8,7 @@ from ._axis_nan_policy import _axis_nan_policy_factory
 
 
 @_axis_nan_policy_factory(
-    lambda x: x, n_outputs=1, result_to_tuple=lambda x: (x,)
+    lambda x: x, n_outputs=1, result_to_tuple=lambda x, _: (x,)
 )
 def variation(a, axis=0, nan_policy='propagate', ddof=0, *, keepdims=False):
     """

